L1Trigger/L1THGCal/test/HGCalTriggerGeomTesterV9.cc

```diff
@@ -0,0 +1,1045 @@
+    treeCells_->Branch("layer"          , &cellLayer_         , "layer/I");
+    treeCells_->Branch("waferu"          , &cellWaferU_         , "waferu/I");
+    treeCells_->Branch("waferv"          , &cellWaferV_         , "waferv/I");
+    treeCells_->Branch("wafertype"      , &cellWaferType_     , "wafertype/I");
+    treeCells_->Branch("waferrow"          , &cellWaferRow_         , "waferrow/I");
+    treeCells_->Branch("wafercolumn"          , &cellWaferColumn_         , "wafercolumn/I");
+    treeCells_->Branch("cellu"           , &cellU_              , "cellu/I");
+    treeCells_->Branch("cellv"           , &cellV_              , "cellv/I");
+    treeCells_->Branch("x"              , &cellX_             , "x/F");
+    treeCells_->Branch("y"              , &cellY_             , "y/F");
+    treeCells_->Branch("z"              , &cellZ_             , "z/F");
+    treeCells_->Branch("corner_n"       , &cellCornersN_     , "corner_n/I");
+    treeCells_->Branch("corner_x"       , cellCornersX_.get()      , "corner_x[corner_n]/F");
+    treeCells_->Branch("corner_y"       , cellCornersY_.get()      , "corner_y[corner_n]/F");
+    treeCells_->Branch("corner_z"       , cellCornersZ_.get()      , "corner_z[corner_n]/F");
+    //
+    treeCellsBH_ = fs_->make<TTree>("TreeCellsBH","Tree of all BH cells");
+    treeCellsBH_->Branch("id", &cellBHId_, "id/I");
+    treeCellsBH_->Branch("type", &cellBHType_, "type/I");
+    treeCellsBH_->Branch("zside", &cellBHSide_, "zside/I");
+    treeCellsBH_->Branch("subdet", &cellBHSubdet_, "subdet/I");
+    treeCellsBH_->Branch("layer", &cellBHLayer_, "layer/I");
+    treeCellsBH_->Branch("ieta", &cellBHIEta_, "ieta/I");
+    treeCellsBH_->Branch("iphi", &cellBHIPhi_, "iphi/I");
+    treeCellsBH_->Branch("eta", &cellBHEta_, "eta/F");
+    treeCellsBH_->Branch("phi", &cellBHPhi_, "phi/F");
+    treeCellsBH_->Branch("x", &cellBHX_, "x/F");
+    treeCellsBH_->Branch("y", &cellBHY_, "y/F");
+    treeCellsBH_->Branch("z", &cellBHZ_, "z/F");
+    treeCellsBH_->Branch("x1", &cellBHX1_, "x1/F");
+    treeCellsBH_->Branch("y1", &cellBHY1_, "y1/F");
+    treeCellsBH_->Branch("x2", &cellBHX2_, "x2/F");
+    treeCellsBH_->Branch("y2", &cellBHY2_, "y2/F");
+    treeCellsBH_->Branch("x3", &cellBHX3_, "x3/F");
+    treeCellsBH_->Branch("y3", &cellBHY3_, "y3/F");
+    treeCellsBH_->Branch("x4", &cellBHX4_, "x4/F");
+    treeCellsBH_->Branch("y4", &cellBHY4_, "y4/F");
+}
+
+
+
+/*****************************************************************/
+HGCalTriggerGeomTesterV9::~HGCalTriggerGeomTesterV9() 
+/*****************************************************************/
+{
+}
+
+/*****************************************************************/
+void HGCalTriggerGeomTesterV9::beginRun(const edm::Run& /*run*/, 
+                                          const edm::EventSetup& es)
+/*****************************************************************/
+{
+    es.get<CaloGeometryRecord>().get("", triggerGeometry_);
+
+    try
```

Comment:
it would be better to have these internal functions return true or false to denote errors, rather than using try/catch